bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -0,0 +1,913 @@
+    }
+
+    public void initialize(@Nullable String authstring, @Nullable String pinCode, @Nullable String userName) {
+        logger.debug("VerisureSession:initialize");
+        if (authstring != null) {
+            this.authstring = authstring.substring(0);
+            this.pinCode = pinCode;
+            this.userName = userName;
+            // Try to login to Verisure
+            if (logIn()) {
+                getInstallations();
+            } else {
+                logger.warn("Failed to login to Verisure!");
+            }
+        }
+    }
+
+    public boolean refresh() {
+        int statusCode = areWeLoggedIn();
+        if (statusCode == 503) {
+            return false;
+        } else if (!areWeLoggedOut && (statusCode == 1)) {
+            updateStatus();
+            return true;
+        } else {
+            if (logIn()) {
+                updateStatus();
+                areWeLoggedOut = false;
+                return true;
+            } else {
+                areWeLoggedOut = true;
+                return false;
+            }
+        }
+    }
+
+    public int sendCommand(String url, String data, @Nullable BigDecimal installationId) {
+        logger.debug("Sending command with URL {} and data {}", url, data);
+        configureInstallationInstance(installationId);
+        int httpResultCode = setSessionCookieAuthLogin();
+        if (httpResultCode == HttpStatus.OK_200) {
+            return sendHTTPpost(url, data);
+        } else {
+            return httpResultCode;
+        }
+    }
+
+    public boolean unregisterDeviceStatusListener(DeviceStatusListener deviceStatusListener) {
+        logger.debug("unregisterDeviceStatusListener for listener {}", deviceStatusListener);
+        return deviceStatusListeners.remove(deviceStatusListener);
+    }
+
+    public boolean registerDeviceStatusListener(DeviceStatusListener deviceStatusListener) {
+        logger.debug("registerDeviceStatusListener for listener {}", deviceStatusListener);
+        return deviceStatusListeners.add(deviceStatusListener);
+    }
+
+    public void dispose() {
+    }
+
+    public @Nullable VerisureThingJSON getVerisureThing(String deviceId) {
+        return verisureThings.get(deviceId);
+    }
+
+    public HashMap<String, org.openhab.binding.verisure.internal.model.VerisureThingJSON> getVerisureThings() {
+        return verisureThings;
+    }
+
+    public @Nullable String getCsrf() {
+        return csrf;
+    }
+
+    public @Nullable String getPinCode() {
+        return pinCode;
+    }
+
+    public String getApiServerInUse() {
+        return apiServerInUse;
+    }
+
+    public void setApiServerInUse(String apiServerInUse) {
+        this.apiServerInUse = apiServerInUse;
+    }
+
+    public String getOtherApiServer() {
+        for (String apiServer : APISERVERLIST) {
+            if (!apiServer.equals(apiServerInUse)) {
+                return apiServer;
+            }
+        }
+        return apiServerInUse;
+    }
```

Comment:
This doesn't work, if we need to add third or more api servers on the list.

Reply:
True, I'll fix!

Reply:
I've now updated the binding according to the latest review comments from @paulianttila.

Reply:
@Hilbrand @paulianttila Can the cause of the Travis build failures be that I need to sync master to my branch? It does not fail due to the Verisure binding what I can see.

Reply:
ping @paulianttila @Hilbrand

Reply:
Yes that is possible the cause. You can get the updated source by rebasing with `git pull --rebase` and than push with force to make sure you get the sources back: `git push --force-with-lease`

Reply:
@Hilbrand @paulianttila Is there anything that I'm suppose to do for the moment?